test.py

```diff
@@ -105,6 +105,47 @@ def test_general_case_integers(self):
 
         self.assertTrue(stack.is_empty())
 
+    def test_push_peek_jz(self):
+        stack = StringStack(initial_size=5)
+        self.assertTrue(stack.is_empty())
+        self.assertRaises(SizeError, stack.peek)
+
+        stack.push("Item 1")
+        self.assertEqual(stack.size, 1)
+        self.assertEqual(stack.peek(), "Item 1")
+        self.assertFalse(stack.is_empty())
+
+        stack.push("Item 2")
+        stack.push("Item 3")
+        self.assertEqual(stack.size, 3)
+        self.assertEqual(stack.peek(), "Item 3")
+
+        stack.push("Item 4")
+        stack.push("Item 5")
+        self.assertRaises(SizeError, stack.push, "Item 6")
+
+    def test_pop_jz(self):
+        stack = IntegerStack(initial_size=5)
```

Comment:
Hmm we've been working with fairly small numbers, but just for funsies, what if we set the initial_size to be like 10_000? You may need to change the rest of the test slightly, but should be fairly simple to do so